h5py/_hl/selections.py

```diff
@@ -91,6 +91,68 @@ def select(shape, args, dsid):
     return sel
 
 
+class MultiBlockSlice(object):
+
+    """
+        A conceptual extension of the built-in slice object to allow selections
+        using start, stride, count and block.
+
+        If given, these parameters will be passed directly to
+        H5Sselect_hyperslab. The defaults are start=0, stride=1, block=1,
+        count=length, which will select the full extent.
+
+        __init__(start, count, stride, block) => Create a new MultiBlockSlice, storing
+            any given selection parameters and using defaults for the others
+        start => The offset of the starting element of the specified hyperslab
+        count => The number of blocks to select
+        stride => The number of elements between the start of one block and the next
+        block => The number of elements in each block
+
+    """
+
+    def __init__(self, start=0, count=None, stride=1, block=1):
+        if start < 0:
+            raise ValueError("Start must be positive")
```

Comment:
```suggestion
            raise ValueError("Start can't be negative")
```

It can be 0, which isn't positive. ;-) This is nitpicking, but I wouldn't want to upset the mathematicians.